src/go/rpk/pkg/cli/cmd/generate/prometheus.go

```diff
@@ -206,3 +228,20 @@ func splitAddress(address string) (string, int, error) {
 	}
 	return host, port, nil
 }
+
+func createClient(fs afero.Fs, addr string, tlsCfg tlsConfig) (*kgo.Client, error) {
```

Comment:
+1 for this PR, but I think we should have a follow up issue to convert all areas that create the Kafka client's to the one `kafka` package.

Reply:
https://github.com/redpanda-data/redpanda/issues/7192